src/foam/u2/Accordion.js

```diff
@@ -0,0 +1,247 @@
+    }
+    ^.expanded .foam-u2-ActionView-toggle {
+      transform: rotate(90deg);
+      transition: transform 0.3s;
+    }
+  `,
+
+  properties: [
+    'title',
+    {
+      name: 'icon',
+      value: null,
+      documentation: `
+        An icon that appears before the title in the Accordion toolbar.
+
+        Example:
+          this.start(foam.u2.Accordion, {
+            title: 'Action Accordion',
+            icon: {
+              class: 'foam.u2.tag.Image',
+              data: 'images/my-icon.svg'
+            },
+          })
+            .add('This is hidden content...');
+      `,
+    },
+    {
+      name: 'actions',
+      value: [],
+      documentation: `
+        A list of actions that appear in the accordion toolbar.
+
+        Example:
+          this.start(foam.u2.Accordion, {
+            title: 'Action Accordion',
+            actions: [
+              {
+                id: 'unique-action-1',  // Must be globally unique.
+                name: 'action-1',  // Can be shared between different accordions for styling purposes, generates a classname.
+                label: 'Action 1',
+                tooltip: 'Action 1 Tip.',
+                icon: 'images/my-action-icon.svg',
+                hoverColor: 'red',
+                action: this.someFunction
+              },
+              {
+                label: 'Action 2',
+                action: this.someOtherFunction
+              }
+            ]
+          })
+            .add('This is hidden content...');
+      `
+    },
+    {
+      class: 'String',
+      name: 'expandIconPosition',
+      view: {
+        class: 'foam.u2.view.ChoiceView',
+        choices: [ 'left', 'right' ]
+      },
+      value: 'right'
+    },
+    {
+      class: 'Boolean',
+      name: 'expanded',
+      value: true
+    }
+  ],
+
+  methods: [
+    function init() {
+
+      const self = this;
+
+      this
+        .addClass()
+        .enableClass('expanded', this.expanded$)
+        .start('div')
+          .addClass(this.myClass('toolbar'))
+          .on('click', this.onToggle)
+          .start('div')
+            .addClass(this.myClass('title'))
+            .callIf(this.expandIconPosition === 'left', function() {
+              this.start(
+                self.TOGGLE
+              ).addClass(self.myClass('toggle'))
+              .on('click', self.onToggle);
+            })
+            .callIf(this.icon, function() {
+              this.tag(self.icon);
+            })
+            .start('span')
+              .addClass(this.myClass('title'))
+              .add(this.title$)
+            .end()
+          .end()
+          .start()
+            .addClass(this.myClass('actions'))
+            .forEach(this.actions, function(action) {
+              const actionId = `${self.id}-${action.name}`;
+              this.start(self.ActionView, {
+                id: actionId,
+                action: action.action,
+                data: action.data || self,
```

Comment:
Do actions have data?

Reply:
What you are looking for are called ActionReference, they can store an Action and data that the action should be associated with. Then you can simply call .start(youActionReference)